t2.go

```diff
@@ -0,0 +1,88 @@
+package ftdc
+
+import (
+	"context"
+	"io"
+	"log"
+	"math"
+
+	"github.com/evergreen-ci/birch"
+	"github.com/pkg/errors"
+)
+
+const SECOND_MS int64 = 1000
+const MAX_SAMPLES = 300
+
+func TranslateGenny(ctx context.Context, iter *ChunkIterator, output io.Writer, actorOpName string) error {
+	collector := NewStreamingCollector(MAX_SAMPLES, output)
+	var err error
+	defer func() {
+		err = FlushCollector(collector, output)
+	}()
+
+	currentSecond := int64(0)
+	endOfSecondIdx := -1
+	prevChunk := iter.Chunk()
```

Comment:
I think `iter.Chunk()` will return `nil` until you call `iter.Next()`.

Reply:
I resolve this by checking if prevChunk is nil, then assign to iter.Chunk() in lines 35-37 with:

```
if prevChunk == nil {
	prevChunk = iter.Chunk()
}
```

Reply:
Then I would do `var prevChunk *ftdc.Chunk`